File: src/main/java/com/muxiu1997/sharewhereiam/command/CommandError.java
package com.muxiu1997.sharewhereiam.command;

import net.minecraft.command.CommandException;

import javax.annotation.Nonnull;

public enum CommandError {
    INVALID_WAYPOINT_CODE,
    ;

    @Nonnull
    public CommandException exception() {
        return new CommandException(LOCALISATION_KEY_PREFIX + name());
    }

    public static final String LOCALISATION_KEY_PREFIX = "sharewhereiam.commanderror.";
}

File: src/main/kotlin/com/muxiu1997/sharewhereiam/command/CommandError.kt
package com.muxiu1997.sharewhereiam.command

import com.muxiu1997.sharewhereiam.localization.Lang
import net.minecraft.command.CommandException

enum class CommandError(private val localizationKey: String) {
    INVALID_WAYPOINT_CODE(Lang.ERROR_INVALID_WAYPOINT_CODE.key);

    fun exception(): CommandException {
        return CommandException(localizationKey)
    }
}
